### PR TITLE
Version fields for inner repositories are now set properly.

### PR DESCRIPTION
**Pull request by cengizkandemir**
This pull request addresses the Issue #2.

Implementation:
- Set the static project path field of "Git" using project.projectDir.absolutePath.
- Run a dummy git command using the supplied path. Run the original git commands on failure to find the git repository.

### CODE REVIEW

**Comment by rockerhieu** [comment]
I would prefer to add a new attribute `gitDir` for `versionberg`, e.g:

```groovy
versionberg {
    major 1
    minor 0
    gitDir $projectDir // then use this value for the git command
}
```

I think this approach will be more flexible and give the developers full controls on choosing which git repo they want to use to manage the version.

**Comment by cengizkandemir** [comment]
I've added "gitDir" field. Setting it as
```
versionberg {
    gitDir "${project.projectDir.absolutePath}/.."
    ...
}
```

did the trick. Can you please review the commit?

File: versionberg/src/main/groovy/io/github/rockerhieu/versionberg/Git.groovy [hunk 1]
```diff
@@ -26,9 +26,12 @@ package io.github.rockerhieu.versionberg
  * Created by rockerhieu on 9/12/16.
  */
 class Git {
+    static String repositoryPath = ''
+
     static String getCommitSha() {
         try {
-            return 'git rev-parse --short HEAD'.execute().text.trim()
+            def externalPathOption = checkInnerGitRepo() ? "--git-dir=${repositoryPath}/.git" : ''
```
Comment (rockerhieu) [inline versionberg/src/main/groovy/io/github/rockerhieu/versionberg/Git.groovy]: `checkInnerGitRepo` is unnecessary as `getCommitSha` can just accept `gitDir` as a parameter

Comment (rockerhieu) [reply versionberg/src/main/groovy/io/github/rockerhieu/versionberg/Git.groovy]: To be clear, here you can always use `gitDir` in the git command.

File: versionberg/src/main/groovy/io/github/rockerhieu/versionberg/Git.groovy [hunk 2]
```diff
@@ -26,9 +26,12 @@ package io.github.rockerhieu.versionberg
  * Created by rockerhieu on 9/12/16.
  */
 class Git {
+    static String repositoryPath = ''
```
Comment (rockerhieu) [inline versionberg/src/main/groovy/io/github/rockerhieu/versionberg/Git.groovy]: I think we should avoid using static stateful variable. How about `getCommitSha(gitDir)` and `getCommitCount(gitDir)`?

File: versionberg/src/main/groovy/io/github/rockerhieu/versionberg/Git.groovy [hunk 3]
```diff
@@ -50,4 +54,16 @@ class Git {
         }
         return 0
     }
+    private static boolean checkInnerGitRepo() {
```
Comment (rockerhieu) [inline versionberg/src/main/groovy/io/github/rockerhieu/versionberg/Git.groovy]: Using this method is unnecessary. If `gitDir` is undefined, assign the root project git directory into it, then always use `gitDir` in all git commands.

**Review by rockerhieu — CHANGES_REQUESTED** [review]
Using `checkInnerGitRepo` method is unnecessary. If `gitDir` is undefined, assign the root project git directory into it, then always use `gitDir` in all git commands.

Avoid using static variables, i.e `getCommitSha(gitDir)` and `getCommitCount(gitDir)` are more preferable.

**Comment by cengizkandemir** [comment]
That was my initial plan but had some problems making it work. I will try to refactor tomorrow.

**Comment by cengizkandemir** [comment]
I've made the necessary changes. There was a problem with printing version information too early, that is, printing before the gitDir field is set in some subproject. It prints the hash and commit count of root directory. I solved it by printing after evaluation. Not sure if that's best hook, however.

**Review by rockerhieu — APPROVED** [review]

**Comment by rockerhieu** [comment]
Looking good, I will do a final test in the next couple hours and let you know.

**Comment by rockerhieu** [comment]
I will do some more improvements on my end, then I will publish a new version soon. Thanks for your contribution @cengizkandemir.

**Comment by cengizkandemir** [comment]
Thanks. I think you can also close the Issue #2 .

**Comment by rockerhieu** [comment]
Available in v1.0.1

**Pull request merged**